src/main/java/ch/fhnw/algd1/converters/utf8/UTF8Converter.java

```diff
@@ -3,37 +3,167 @@
-		else return false;
-	}
-
-	private static boolean isFollowup(byte b) {
-		return (b & 0b1100_0000) == 0b1000_0000;
-	}
+
+    // big if else. but I could not be bothered
+    public static byte[] codePointToUTF(int x) {
+        byte[] b;
+
+        String binaryString = intToBinaryString(x);
+
+        if (binaryString.length() <= 7) {
+            b = new byte[1];
+            b[0] = (byte) binaryStringToInt(binaryString);
+
+        } else if (binaryString.length() <= 11) {
+            b = new byte[2];
+
+            String followUp = binaryString.substring(binaryString.length() - 6);
+            String main = binaryString.substring(0, binaryString.length() - 6);
+
+            b[1] = (byte) binaryStringToInt("10" + followUp);
+
+            int padding = 5 - main.length();
+            main = "110" + "0".repeat(padding) + main;
+            b[0] = (byte) binaryStringToInt(main);
+
+        } else if (binaryString.length() <= 16) {
+            b = new byte[3];
+
+            String followUpPart = binaryString.substring(binaryString.length() - 12);
+            b[2] = (byte) binaryStringToInt("10" + followUpPart.substring(followUpPart.length() - 6));
+            b[1] = (byte) binaryStringToInt("10" + followUpPart.substring(0, followUpPart.length() - 6));
+
+            String main = binaryString.substring(0, binaryString.length() - 12);
+            int padding = 4 - main.length();
+            main = "1110" + "0".repeat(padding) + main;
+            b[0] = (byte) binaryStringToInt(main);
+
+        } else if (binaryString.length() <= 21) {
+            b = new byte[4];
+
+            if (binaryString.length() == 17) {
+                binaryString = "0" + binaryString;
+            }
+
+            String followUpPart = binaryString.substring(binaryString.length() - 18);
+            String main = binaryString.substring(0, binaryString.length() - 18);
+
+            b[3] = (byte) binaryStringToInt("10" + followUpPart.substring(followUpPart.length() - 6));
+            b[2] = (byte) binaryStringToInt("10" + followUpPart.substring(6, binaryString.length() - 6));
+            b[1] = (byte) binaryStringToInt("10" + followUpPart.substring(0, 6));
+
+            int padding = 3 - main.length();
+            main = "11110" + "0".repeat(padding) + main;
+            b[0] = (byte) binaryStringToInt(main);
+
+        } else {
+            return new byte[]{0};
+        }
+
+        return b;
+    }
+
+    public static int UTFtoCodePoint(byte[] bytes) {
+        if (isValidUTF8(bytes)) {
+            int countBytes = bytes.length;
+            int result = 0;
+
+            if (countBytes == 1) {
+                System.out.println("1 bytes");
+                String mainBinaryString = intToBinaryString(bytes[0]);
+                //intToBinaryString does not give leading "0" chars.
+                // so we do not need to trim it
+                return binaryStringToInt(mainBinaryString);
```

Comment:
Hier wäre natürlich eine kurze Fassung denkbar:
```suggestion
                return bytes[0];
```